tests/components/image/test_init.py

```diff
@@ -0,0 +1,76 @@
+"""Test that we can upload images."""
+import pathlib
+import tempfile
+
+from aiohttp import ClientSession, ClientWebSocketResponse
+
+from homeassistant.components.websocket_api import const as ws_const
+from homeassistant.setup import async_setup_component
+from homeassistant.util import dt as util_dt
+
+from tests.async_mock import patch
+
+
+async def test_upload_image(hass, hass_client, hass_ws_client):
+    """Test we can upload an image."""
+    now = util_dt.utcnow()
+    test_image = pathlib.Path(__file__).parent / "logo.png"
+
+    with tempfile.TemporaryDirectory() as tempdir, patch.object(
```

Comment:
Pytest has a `tmp_path` fixture:
https://docs.pytest.org/en/stable/tmpdir.html#the-tmp-path-fixture